homeassistant/components/frontier_silicon/config_flow.py

```diff
@@ -0,0 +1,235 @@
+        except InvalidPinException:
+            return self.async_abort(reason="invalid_auth")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        # For manually added devices the unique_id is the radio_id,
+        # for devices discovered through SSDP it is the UDN
+        await self.async_set_unique_id(self._unique_id, raise_on_progress=False)
+        self._abort_if_unique_id_configured()
+
+        self._name = import_info[CONF_NAME] or "Radio"
+
+        _LOGGER.warning("Frontier Silicon %s imported from YAML config", self._name)
+        return await self._create_entry(pin=import_info[CONF_PIN])
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step of manual configuration."""
+        errors = {}
+
+        if user_input:
+            device_url = (
+                f"http://{user_input[CONF_HOST]}:{user_input[CONF_PORT]}/device"
+            )
+            try:
+                self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+            except FSConnectionError:
+                errors["base"] = "cannot_connect"
+            except Exception as exception:  # pylint: disable=broad-except
+                _LOGGER.exception(exception)
+                errors["base"] = "unknown"
+            else:
+                return await self._async_step_device_config_if_needed()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowResult:
+        """Process entity discovered via SSDP."""
+
+        device_url = discovery_info.ssdp_location
+
+        speaker_name = discovery_info.ssdp_headers.get(SSDP_ATTR_SPEAKER_NAME)
+        self.context["title_placeholders"] = {"name": speaker_name}
+
+        try:
+            self._webfsapi_url = await AFSAPI.get_webfsapi_endpoint(device_url)
+        except FSConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except Exception as exception:  # pylint: disable=broad-except
+            _LOGGER.exception(exception)
+            return self.async_abort(reason="unknown")
+
+        # For manually added devices the unique_id is the radio_id,
+        # for devices discovered through SSDP it is the UDN
+        self._unique_id = discovery_info.ssdp_udn
+        await self.async_set_unique_id(self._unique_id)
+        self._abort_if_unique_id_configured(
+            updates={CONF_WEBFSAPI_URL: self._webfsapi_url}, reload_on_update=True
+        )
+
+        return await self._async_step_device_config_if_needed(show_confirm=True)
+
+    async def _async_step_device_config_if_needed(
+        self, show_confirm=False
+    ) -> FlowResult:
```

Comment:
`show_confirm` is not needed at this stage. It can be added later when SSDP is implemented.

```suggestion
    async def _async_step_device_config_if_needed(
        self
    ) -> FlowResult:
```